src/HookManager.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#include "HookManager.h"
+
+#include <QRegExp>
+
+#include "HookFormatter.h"
+
+HookManager::HookManager(QObject* parent)
+    : QObject(parent),
+    m_event(),
+    m_eventSpec(),
+    m_hooksSignedUp()
+{
+}
+
+
+void HookManager::connectHook(QByteArray eventSpec, HookFormatter* formatter)
+{
+    /// Remove keyword before key sequence, indicated by colon. At the moment,
+    /// everything up to (and including) the colon is removed and discarded, 
+    /// so not specifying 'key:' works as well.
+    QString specifier = QString::fromUtf8(eventSpec);
+    //specifier.count(QRegExp("[:]"));
+    specifier.replace(QRegExp(".*[:]"), "");
+    QKeySequence requested = QKeySequence::fromString(specifier, QKeySequence::PortableText);
+
+    /// Go through all shortcuts, see if one exists, if not create it and connect it to HookFormatter
+    int numberEvents = m_event.count();
+    bool taken = false;
+    for(int i=0; i<numberEvents; i++)
+    {
+        if(m_eventSpec.at(i)==requested)
```

Comment:
I'd suggest the unchecked `m_eventSpecs[i]` rather than using `at()`, for consistency with all the other code.  This goes for most uses of `at()` really.

The style I tend to follow is to check lengths explicitly in those places which need it, and avoid the overhead elsewhere.